inference/worker/chat_chain_utils.py

```diff
@@ -293,34 +296,27 @@ def func(req):
 
         param_location = endpoint.params[0].in_ if len(endpoint.params) > 0 else "query"
 
-        # some plugins do not have operation_id, so we use path as fallback
+        # If plugin has no operation_id, use path as fallback
         path = endpoint.path[1:] if endpoint.path and len(endpoint.path) > 0 else endpoint.path
 
         tool = Tool(
             name=endpoint.operation_id if endpoint.operation_id != "" else path,
-            # Could be path, e.g /api/v1/endpoint
-            # but it can lead LLM to makeup some URLs
-            # and problem with EP description is that
-            # it can be too long for some plugins
+            # Could be path, e.g /api/v1/endpoint but can lead LLM to invent URLs
+            # Problem with EP description is that it is too long for some plugins
             func=create_tool_func(endpoint, param_location),
             description=f"{openapi_specification_title}{parameters_description}{payload_description}tool description: {endpoint.summary}\n",
         )
         tools.append(tool)
 
     tools_string = "\n".join([f"> {tool.name}{tool.description}" for tool in tools])
-    # NOTE: This can be super long for some plugins, that I tested so far.
-    # and because we don't have 32k CTX size, we need to truncate it.
-    plugin_description_for_model = truncate_str(llm_plugin["description_for_model"], 512)
+    # This can be long for some plugins, we need to truncate due to ctx limitations
+    plugin_description_for_model = truncate_str(llm_plugin.description_for_model, 512)
```

Comment:
we could extract 512 to some constant at the top of the file(DESCRIPTION_FOR_MODEL_MAX_LENGTH = 256 ), maybe even reduce it to 256 for now, I forgot that part on my end. sorry...

Reply:
Makes sense, done